src/Demo.java

```diff
@@ -4,9 +4,9 @@ public class Demo {
 
     public static void main(String[] args) {
 
-        PrefixTree prefixTree = new PrefixTree();
-        Node root = new Node();
-        prefixTree.setRoot(root);
+
+        Node root = new Node('\u0000');
+        PrefixTree prefixTree = new PrefixTree(root);
```

Comment:
The root node should be a concern of the tree class really. The consumer of the class shouldn't care about it.